helper/resource/teststep_providers_test.go

```diff
@@ -1778,6 +1782,54 @@ func TestTest_TestStep_ProviderFactories_Import_External_WithPersistMatch(t *tes
 	})
 }
 
+func TestTest_TestStep_ProviderFactories_Import_External_WithPersistMatch_WithPersistWorkingDir(t *testing.T) {
+	var result1, result2 string
+
+	t.Setenv(plugintest.EnvTfAccPersistWorkingDir, "1")
+	workingDir := t.TempDir()
+
+	testSteps := []TestStep{
+		{
+			Config:             `resource "random_password" "test" { length = 12 }`,
+			ResourceName:       "random_password.test",
+			ImportState:        true,
+			ImportStateId:      "Z=:cbrJE?Ltg",
+			ImportStatePersist: true,
+			ImportStateCheck: composeImportStateCheck(
+				testExtractResourceAttrInstanceState("none", "result", &result1),
+			),
+		},
+		{
+			Config: `resource "random_password" "test" { length = 12 }`,
+			Check: ComposeTestCheckFunc(
+				testExtractResourceAttr("random_password.test", "result", &result2),
+				testCheckAttributeValuesEqual(&result1, &result2),
+			),
+		},
+	}
+
+	Test(t, TestCase{
+		ExternalProviders: map[string]ExternalProvider{
+			"random": {
+				Source: "registry.terraform.io/hashicorp/random",
+			},
+		},
+		WorkingDir: workingDir,
+		Steps:      testSteps,
+	})
+
+	workingDirPath := filepath.Dir(workingDir)
+
+	for k := range testSteps {
+		dir := workingDirPath + "_" + strconv.Itoa(k+1)
+
+		_, err := os.ReadDir(dir)
```

Comment:
Open question: Should we try to verify any of the contents within the directories? I think for at least one of the tests it might be good to verify that we get:

- Step 1: configuration file and state file
- Step 2: configuration file, plan file, and state file

Reply:
I think this is a good shout. I've added some checks to verify that config, state and plan files exist where appropriate.